### PR TITLE
Thread local storage

### PR DESCRIPTION
**Pull request by jprashanth**
In reference to #404 

@TheJJ instead of just failing, i have added fallback for thread_local storage. Now, `openage` compiles with AppleClang.

### CODE REVIEW

File: libopenage/config.h.in [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #ifndef OPENAGE_CONFIG_H_
 #define OPENAGE_CONFIG_H_
 
+#cmakedefine01 HAVE_THREAD_LOCAL_STORAGE
```
Comment (TheJJ) [inline libopenage/config.h.in]: Why use the `#cmakedefine01`? The other definitions below use a standard `#define`.

Comment (jprashanth) [inline libopenage/config.h.in]: No major reason. I wasn't sure what `check_cxx_source_compiles` returned. Also, `cmakedefine01` handles all the cmake's true/false cases

File: libopenage/util/thread_id.cpp [hunk 2]
```diff
@@ -2,18 +2,52 @@
...
 
-const thread_local ThreadIdSupplier current_thread_id;
+#endif
 
+size_t get_current_thread_id() {
+	#if HAVE_THREAD_LOCAL_STORAGE
+	static thread_local ThreadIdSupplier current_thread_id;
+	return current_thread_id.val;
+	#else
+	return std::hash<std::thread::id>()(std::this_thread::get_id());
```
Comment (TheJJ) [inline libopenage/util/thread_id.cpp]: Are the thread ids really unique? if they are, why are we hashing again, the entropy doesn't increase.

Comment (jprashanth) [inline libopenage/util/thread_id.cpp]: thread ids using `std::thread::id` are unique at any given time. They may not be unique across the lifespan of application.

`std::thread::id` is an object, using hash to get `size_t`.

File: libopenage/config.h.in (hunk 1 above)
Comment (TheJJ) [inline libopenage/config.h.in]: Can you change it to a simple `#define` then please?

File: libopenage/util/thread_id.cpp (hunk 2 above)
Comment (TheJJ) [inline libopenage/util/thread_id.cpp]: Ok, then this workaround should do the right thing probably.

Comment (jprashanth) [inline libopenage/util/thread_id.cpp]: yes

**Pull request merged**